src/nvim/os/env.c

```diff
@@ -243,19 +243,24 @@ void expand_env_esc(char_u *srcp, char_u *dst, int dstlen, bool esc, bool one,
         // Verify that we have found the end of a UNIX ${VAR} style variable
         if (src[1] == '{' && *tail != '}') {
           var = NULL;
-        } else if (src[1] == '{') {
-          ++tail;
-        }
+        } else {
+          if (src[1] == '{') {
```

Comment:
If you look a a couple lines down to [Line 261](https://github.com/neovim/neovim/pull/2250/files#diff-145b92bbfb361cae1385b1efb2dea428R261), you'll see that I'm closing a brace if `MSWIN` or `UNIX` are defined.  

That brace is closing this else block. So that if `MSWIN` or `UNIX` are defined we get to [Line 259](https://github.com/neovim/neovim/pull/2250/files#diff-145b92bbfb361cae1385b1efb2dea428R259) if and only if `var` is not null.